File: C/Stack.c
// Implementation of Stack 
// Operation : Push() and Pop()

#include <stdio.h>
#define MAX 10
int A[10];
void push(int [],int,int*);
int pop(int[],int*);
void main()
{
	int top = 0,ch,num;
	do
	{
	
		printf("\n enter your choices \n1.Push operation \n2.Pop operration \n3.Exit\n");
		scanf("%d",&ch);
		switch(ch)
		{
			case 1:
				printf("\n Pls enter  a num to be pushed : ");
				scanf("%d",&num);
				push(A,num,&top);
				break;
			case 2:
				printf("\n Pls enter a number to be pop:");
				scanf("%d",num);
				break;
			case 3:
				break;
			default:
				printf("\n Wrong choise !");
		}
	}while(ch != 3);
}
void push(int A[],int num, int* top)
{
	if(*top == MAX)
	{
		printf("\n The stack is FULL !");
		return;
	}
	else
	{
		*top = *top + 1;
		A[*top] = num;
		printf("\n The pushed element is %d and position is %d",num,*top);
	}
}
int pop(int A[], int* top)
{
	int num;
	if(*top == 0)
	{
		printf("\n The stack is empty !");
		return;
	}
	else
	{
		A[*top] = num;
		printf("\n The poped element is %d and position is %d",num,*top);
		*top = *top - 1;
		return(num);
	}
}

File: Java/Stack.java
public class Stack<Item> {

   private Item[] a;
   private int n;

   
   public Stack() {
      a = (Item[]) new Object[2];
      n = 0;
   }

  
   public boolean isEmpty() {
      return n == 0;
   }
   
   public void push(Item item) { 
      if (n == a.length)
         resize(2 * a.length);
      a[n++] = item;
   }

   public Item pop() {
      Item item = a[--n];
      if (n > 0 && n == a.length/4)
         resize(a.length/2);
      return item;
   }

   private void resize(int max) {
      Item[] temp;
      temp = (Item[]) new Object[max];
      for (int i = 0; i < n; i++)
         temp[i] = a[i];
      a = temp;
   }
}